src/angularjs/src/app/places/map/places-map.directive.js

```diff
@@ -0,0 +1,121 @@
+(function() {
+
+    /* @ngInject */
+    function PlacesMapController($filter, $http, $sanitize, $scope) {
+        var ctl = this;
+        ctl.map = null;
+        ctl.layerControl = null;
+
+        ctl.$onInit = function () {
+            ctl.mapOptions = {
+                scrollWheelZoom: true
+            };
+
+            // TODO: set center and zoom level by zooming to fit geojson polygon bounds
+            ctl.mapCenter = [39.963277, -75.142971];
+            ctl.baselayer = L.tileLayer(
+                'https://stamen-tiles.a.ssl.fastly.net/toner-lite/{z}/{x}/{y}.png', {
+                    attribution: 'Map tiles by <a href="http://stamen.com">Stamen Design</a>, under <a href="http://creativecommons.org/licenses/by/3.0">CC BY 3.0</a>. Data by <a href="http://openstreetmap.org">OpenStreetMap</a>, under <a href="http://www.openstreetmap.org/copyright">ODbL</a>.',
+                    maxZoom: 18
+                });
+
+            $scope.$watch(function(){return ctl.pfbPlacesMapLayers;}, setLayers);
+        };
+
+        ctl.onMapReady = function (map) {
+            ctl.map = map;
+
+            if (ctl.pfbPlacesMapLayers) {
+                setLayers(ctl.pfbPlacesMapLayers);
+            }
+        };
+
+        function setLayers(layers) {
+            if (!layers) {
+                return;
+            }
+
+            if (!ctl.layerControl) {
+                ctl.layerControl = L.control.layers({'Stamen': ctl.baselayer}, []).addTo(ctl.map);
+            }
+
+            _.map(layers, function(url, metric) {
+                var label = $sanitize(metric.replace(/_/g, ' '));
+                $http.get(url).then(function(response) {
+                    if (response.data && response.data.features) {
+                        var layer = L.geoJSON(response.data, {
+                            onEachFeature: onEachFeature
+                        });
+                        ctl.layerControl.addOverlay(layer, label);
+                    }
+                });
+            });
+
+            function onEachFeature(feature, layer) {
+                // TODO: Style marker and popup
+                layer.on({
+                    click: function () {
+                        if (feature && feature.geometry &&
+                            feature.geometry.coordinates) {
+                            var popup = L.popup()
+                                .setLatLng([
+                                    feature.geometry.coordinates[1],
+                                    feature.geometry.coordinates[0]
+                                ])
+                                .setContent(buildLabel(feature.properties));
+                            ctl.map.openPopup(popup);
+                        }
+                    }
+                });
+            }
+
+            /**
+             Convert GeoJSON properties into HTML snippet for Leaflet popup display.
+             */
+            function buildLabel(properties) {
```

Comment:
Ok for now, depending on how fancy the popup gets we might want to switch to an actual ng template. We have plenty of examples of this in other projects.